WBooks/AddNew/AddNewController.swift

```diff
@@ -7,23 +7,165 @@
+        print(error)
+    }
+    
+    private func setImagePicker() {
+        let tapRecognizer = UITapGestureRecognizer(target: self, action: #selector(imageTapped))
+        tapRecognizer.delegate = self
+        _view.cover.addGestureRecognizer(tapRecognizer)
+        _view.cover.isUserInteractionEnabled = true
+    }
+    
+    @objc private func imageTapped(sender: AnyObject) {
+        let alertController = UIAlertController(title: .none, message: .none, preferredStyle: .actionSheet)
+        
+        //Gallery option
+        let chooseAction = UIAlertAction(title: "Gallery", style: .default) { [weak self] _ in
+            let imagePickerController = UIImagePickerController.imagePicker
+            imagePickerController.delegate = self
+            imagePickerController.sourceType = .photoLibrary
+            self?.present(imagePickerController, animated: true, completion: .none)
+        }
+        //Camera option
+        if UIImagePickerController.isSourceTypeAvailable(.camera) {
+            let takeAction = UIAlertAction(title: "Camera", style: .default) { [weak self] _ in
+                let imagePickerController = UIImagePickerController.imagePicker
+                imagePickerController.delegate = self
+                imagePickerController.sourceType = .camera
+                self?.present(imagePickerController, animated: true, completion: .none)
+            }
+            alertController.addAction(takeAction)
+        }
+        //Cancel option
+        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel, handler: .none)
+        
+        alertController.addAction(chooseAction)
+        alertController.addAction(cancelAction)
+        
+        self.present(alertController, animated: true, completion: nil)
     }
 
-    /*
-    // MARK: - Navigation
+    private func configureNavBar() {
+        self.tabBarController?.setNavigationBarTitle("NAVIGATION_BAR_TITLE_ADD_NEW".localized(), font: UIFont.systemFont(ofSize: 18, weight: UIFont.Weight.medium), color: UIColor.white)
+        //Left notification button
+        self.tabBarController?.navigationItem.leftBarButtonItem = nil
+        //Right search button
+        self.tabBarController?.navigationItem.rightBarButtonItem = nil
+    }
+}
 
-    // In a storyboard-based application, you will often want to do a little preparation before navigation
-    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
-        // Get the new view controller using segue.destination.
-        // Pass the selected object to the new view controller.
+extension AddNewController: UIImagePickerControllerDelegate {
+    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [String: Any]) {
+        _view.cover.image = info[UIImagePickerControllerOriginalImage] as? UIImage
+        dismiss(animated: true, completion: nil)
     }
-    */
+}
+
+extension AddNewController: UINavigationControllerDelegate {
```

Comment:
Why is this required?

Reply:
When I make the AddNewController the image picker delegate, it requires AddNewController to conform to UIImagePickerControllerDelegate and UINavigationControllerDelegate. I put it on a separate extension to conform to the Wolox good practices you mentioned. But I can include it on line 156 with the extension that conforms to UIImagePickerControllerDelegate.

Reply:
Yes, if If you are not adding any function, you could set this with UIImagePickerControllerDelegate